features/cellular/TARGET_DRAGONFLY_MODEM/dragonfly_modem_driver/DragonFlyCellularInterface.h

```diff
@@ -0,0 +1,239 @@
+
+typedef struct {
+    device_type dev;
+    char ccid[20+1];    //!< Integrated Circuit Card ID
+    char imsi[15+1];    //!< International Mobile Station Identity
+    char imei[15+1];    //!< International Mobile Equipment Identity
+    char meid[18+1];    //!< Mobile Equipment IDentifier
+    nsapi_error_t  connection_error_flag;
+    bool ppp_connection_up;
+    radio_access_nwk_type rat;
+    nwk_registration_status reg_status;
+} device_info;
+
+/** DragonFlyCellularInterface class
+ *
+ *  This interface serves as the controller/driver for the MTS
+ *  Dragonfly cellular IoT series.
+ */
+class DragonFlyCellularInterface : public CellularInterface {
+
+public:
+    DragonFlyCellularInterface(bool use_USB, bool debugOn=0);
+    ~DragonFlyCellularInterface();
+
+    /** Set the Cellular network credentials
+     *
+     *  Please check documentation of connect() for default behaviour of APN settings.
+     *
+     *  @param apn      Access point name
+     *  @param uname    optionally, Username
+     *  @param pwd      optionally, password
+     */
+    virtual void set_credentials(const char *apn, const char *uname = 0,
+                                                  const char *pwd = 0);
+
+    /** Set the pin code for SIM card
+     *
+     *  @param sim_pin      PIN for the SIM card
+     */
+    virtual void  set_SIM_pin(const char *sim_pin);
+
+    /** Start the interface
+     *
+     *  Attempts to connect to a Cellular network.
+     *  Application should be aborted if NSAPI_ERROR_AUTH_FAILURE is received which means wrong
+     *  pin for SIM card.
+     *  If continued attempts are made with wrong pin number, SIM card might get locked or blocked
+     *  by the operator depending upon regulations.
+     *
+     *  @param sim_pin     PIN for the SIM card
+     *  @param apn         optionally, access point name
+     *  @param uname       optionally, Username
+     *  @param pwd         optionally, password
+     *  @return            NSAPI_ERROR_OK on success, or negative error code on failure
+     */
+    virtual nsapi_error_t connect(const char *sim_pin, const char *apn = 0,
+                                  const char *uname = 0, const char *pwd = 0);
+
+
+    /** Attempt to connect to the Cellular network
+     *
+     *  Brings up the network interface. Connects to the Cellular Radio
+     *  network and then brings up the underlying network stack to be used
+     *  by the cellular modem over PPP interface.
+     *
+     *  If the SIM requires a PIN, and it is not set/invalid, NSAPI_ERROR_AUTH_ERROR is returned.
+     *  Application must abort if NSAPI_ERROR_AUTH_ERROR is returned.
+     *  For APN setup, default behaviour is to use 'internet' as APN string and assuming no authentication
+     *  is required, i.e., username and password are no set. Optionally, a database lookup can be requested
+     *  by turning on the APN database lookup feature. In order to do so, add 'MBED_CONF_MTS_DRAGONFLY_APN_LOOKUP'
```

Comment:
This seems alright to be a config option, as that APN db is not at all exhaustive, It was a customer demand so they can have it if they want to use it. Otherwise, we have our mbed-os API to set proper credentials.